### PR TITLE
feat(channel): Introduce a 'Channel' class for bi-directional process message channels

### LINKED ISSUE #1: Process Channels
## Abstract

A process channel API to allow a spawned child process from the `little-process-box` module to communicate using a message API. 

## Implementation

Provide a `Channel` class to consume a  [`Process`](https://github.com/little-core-labs/little-process-box#process) and provide an implementation of [simple-message-channels](https://github.com/mafintosh/simple-message-channels) that use the child process  `stdin` and `stdout` streams as the readable and writable streams for [duplexify](https://github.com/mafintosh/duplexify).

Spawned child processes are expected to use the same channel API as the spawn caller.

### PR DESCRIPTION
**Pull request by jwerle**
Fixes #1

### CODE REVIEW

**Comment by jwerle** [comment]
cc @agrathwohl

File: channel.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
+const SimpleMessageChannel = require('simple-message-channels')
+const { EventEmitter } = require('events')
+const duplexify = require('duplexify')
```
Comment (bcomnes) [inline channel.js]: Not used.

Comment (bcomnes) [reply channel.js]:
```suggestion
```

Comment (bcomnes) [reply channel.js]: Also fwiw, Mathias mentioned he avoids duplexity these days.  I don't remember the reason.

Comment (jwerle) [reply channel.js]: oof good to know! just removed it

**Pull request merged**